src/libraries/Microsoft.PowerFx.Core/Public/FileLocation.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Syntax
+{
+    /// <summary>
+    /// File-aware, Multi-line source span with Line and Column. 
+    /// Wheras <see cref="Span"/> is a character span within a single expression.
+    /// </summary>
+    public class FileLocation
+    {
+        public string Filename { get; init; }
+
+        // 1-based index
+        public int LineStart { get; init; }
+
+        // 1-based index. 
+        public int ColStart { get; init; }
+
+        /// <summary>
+        /// Apply a span (which is a character offset within a single expression) 
+        /// to this file offset. 
+        /// </summary>
+        /// <param name="originalText"></param>
+        /// <param name="s2"></param>
+        /// <returns></returns>
+        public FileLocation Apply(string originalText, Span s2)
```

Comment:
>originalText [](http://example.com/codeflow?start=41&length=12)

What is this? Is this expression script?

Reply:
yes